### DISCUSSION TITLE
issue: Error showing from last render

### DISCUSSION THREAD

**Discussion by OldManMeta** [react-hook-form/react-hook-form#8537] [Q&A] [answered]
### Version Number

7.32.2

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-errors-17jxfu

### Steps to reproduce

Repro - a single form with a single Textfield element

I have a formstate errors watch
The textfield is resolving with yup
when useEffect executes on a watch of the field & errors - without any errors in the field, the errors object will contain the last error that was resolved

To check this - open the codesandbox repro
Open the console to see the errors
in the textfield add the 4 character word 'test'
clear the console
add a 5th char to the textfield -> you will now see that the errors object still contains the error from the last change event

### Expected behaviour

The error object should be undefined after the change event on the textfield and the resolver has parsed it

The useEffect should execute under the watch, with an undefined errors object

### What browsers are you seeing the problem on?

Chrome, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
form state update will always be updated at the render phase.

**Reply by OldManMeta** [reply]
@bluebill1049 thanks for your response.

I can appreciate that, but it doesn't mean that it's usable.

I expect that is because under the covers, the error object is stored in state, rather than a ref?

While the !! statement works as a Boolean coercion on the Object{} instruction to determine the presence of an error, it's just a dirty method to determine the existence of error(s) inside the object state.

I'm not saying this is an easy scenario to fix - thanks to everyone for the hard work on what is quite clearly the best form solution in React today.
